src/types/genericalias.rs

```diff
@@ -0,0 +1,67 @@
+use crate::ffi_ptr_ext::FfiPtrExt;
+use crate::{ffi, types::any::PyAnyMethods, Bound, PyAny, Python};
+
+/// Represents a Python [`types.GenericAlias`](https://docs.python.org/3/library/types.html#types.GenericAlias) object.
+///
+/// Values of this type are accessed via PyO3's smart pointers, e.g. as
+/// [`Py<PyGenericAlias>`][crate::Py] or [`Bound<'py, PyGenericAlias>`][Bound].
+#[repr(transparent)]
+pub struct PyGenericAlias(PyAny);
+
+pyobject_native_type!(
```

Comment:
Notice that `types.GenericAlias` can be sub-classed but I didn't quite get the caveats of translating that in PyO3 so I might need some help

Reply:
There are a number of issues related to subtyping types defined in Python, so it may not actually work.

See e.g. https://github.com/PyO3/pyo3/issues/4443, https://github.com/PyO3/pyo3/issues/1647, https://github.com/PyO3/pyo3/issues/1644 and https://github.com/PyO3/pyo3/issues/1637#issuecomment-849444381

Reply:
Well, for the time being `types.GenericAlias` is enough then